### ISSUE TITLE
 Check if Get is initialized

### ISSUE THREAD

**Issue by ignaciolopez12** [jonataslaw/get#121]
**Describe the bug**

In my http handler I check for error 401, redirect it to WelcomeScreen with Get.offAllNamed ('/ welcome).
This method works when the App is running.
The problem is, before starting the App, since while the native SplashScreen is displayed, I check if there is a user logged in, if there is, I send a RefreshToken to my Back-End. If it returns Ok, I renew session, if there is an error 401 (as I mentioned before), try to redirect it to WelcomeScreen, but Get shows an error (I suppose it's because the App has not started yet so the instance is not initialized).
Do I have any way to verify this before redirecting it to WelcomeScreen?
If I don't, I have an exception. Attached code and error message.

**To Reproduce**

HttpHandler.dart

> .........
> case 401:
>         String errorDescription = bodyDetails['error_description'];
>         if (errorDescription.contains('Invalid refresh token (expired)')) {
>          // Here, I should check if Get is initialized ... if I do Get! = Null,
>          // the result is false and it doesn't work
>           Get.offAllNamed(SCREEN_WELCOME);
>           throw UnauthorizedException(message: 'Session vencida (refresh token).',printExc: errorResp);
>         } else {
>           throw UnauthorizedException(printExc: errorResp);
>         }
>         break;
> .........

Console error:

> Unhandled exception:
> NoSuchMethodError: The method 'global' was called on null.
> Receiver: null
> Tried calling: global(null)
> #0      Object.noSuchMethod (dart:core-patch/object_patch.dart:53:5)
> #1      Get.offAllNamed (package:get/src/get_main.dart:138:17)
> 

**Flutter Version:**
1.17.1

**Get Version:**
2.5.0

**Describe on which device you found the bug:**
Moto G8 - Android.

**Comment by jonataslaw** [comment]
If you insert it into your main():
main(){
final g = Get ();
runApp()
}


is there an error?

Generally Get is started when GetMaterialApp is read, I didn't understand how the situation occurs, but you can insert this Get in the main method to guarantee the Get start even without the MaterialApp having been called.

**Comment by ignaciolopez12** [comment]
> If you insert it into your main():
> main(){
> final g = Get ();
> runApp()
> }
> 
> is there an error?
> 
> Generally Get is started when GetMaterialApp is read, I didn't understand how the situation occurs, but you can insert this Get in the main method to guarantee the Get start even without the MaterialApp having been called.


Thanks, that has worked. I have checked it with Get.routing! = Null, and before the main RunApp I start the instance Get ().
I ask you, are there cases where there are already APPs working with this Get package? The app I am developing is going to grow a lot and it is for an important company, I hope that this great package does not fail me throughout the development! My question (fear) is if it will remain stable during all these years.

**Comment by jonataslaw** [comment]
> > If you insert it into your main():
> > main(){
> > final g = Get ();
> > runApp()
> > }
> > is there an error?
> > Generally Get is started when GetMaterialApp is read, I didn't understand how the situation occurs, but you can insert this Get in the main method to guarantee the Get start even without the MaterialApp having been called.
> 
> Thanks, that has worked. I have checked it with Get.routing! = Null, and before the main RunApp I start the instance Get ().
> I ask you, are there cases where there are already APPs working with this Get package? The app I am developing is going to grow a lot and it is for an important company, I hope that this great package does not fail me throughout the development! My question (fear) is if it will remain stable during all these years.

1 multinational, 63 web projects for stores in a shopping center. more than 100 mobile projects. 1 social network.
This library is over a year old, and has already spearheaded all of my company's projects. All of our projects need it, so it will continue to be updated, until Flutter no longer exists.

**Comment by ignaciolopez12** [comment]
> > > If you insert it into your main():
> > > main(){
> > > final g = Get ();
> > > runApp()
> > > }
> > > is there an error?
> > > Generally Get is started when GetMaterialApp is read, I didn't understand how the situation occurs, but you can insert this Get in the main method to guarantee the Get start even without the MaterialApp having been called.
> > 
> > 
> > Thanks, that has worked. I have checked it with Get.routing! = Null, and before the main RunApp I start the instance Get ().
> > I ask you, are there cases where there are already APPs working with this Get package? The app I am developing is going to grow a lot and it is for an important company, I hope that this great package does not fail me throughout the development! My question (fear) is if it will remain stable during all these years.
> 
> 1 multinational, 63 web projects for stores in a shopping center. more than 100 mobile projects. 1 social network.
> This library is over a year old, and has already spearheaded all of my company's projects. All of our projects need it, so it will continue to be updated, until Flutter no longer exists.

Very good !! I like this package. As I am, I ask you .. I have a simple transition from WelcomeScreen -> LoginScreen -> HomeScreen, and seeing the log terminal this appears:

Note: when I navigate from login to home, I remove all other routes from the stack. Get.offAllNamed(SCREEN_HOME)

> I/flutter (16630): [GOING TO ROUTE] /
> I/flutter (16630): [GOING TO ROUTE] /login
> I/flutter (16630): [GOING TO ROUTE] null
> I/flutter (16630): [GOING TO ROUTE] /home
> I/flutter (16630): [REMOVING ROUTE] null

Put a breakpoint on all "Get" calls, but I didn't see anything weird. Why do I see null?

**Comment by jonataslaw** [comment]
Everything happens normally, sometimes Flutter does some funny operations to reproduce something simple, and in the Log I put everything literally. Any behavior is displayed on the route. strangely it seems that Flutter creates a temporary route when Get.offAllNamed is called, to remove the previous ones, and display the animation of the routes at the same time. I'm not sure about that, but I'll try to filter the logs to show only real operations.
What you saw seems to be a placebo, it shouldn't influence anything.

**Comment by jonataslaw** [comment]
Based on the previous comment I am referring to the issue, I am closing this.
As for the comment on null routes, I was unable to reproduce the problem using version 2.5.9.

If you face any other problem with Get, do not hesitate to open another issue.
